src/ShareRoutes.jsx

```diff
@@ -32,6 +32,43 @@ function Forward({appPrefix}) {
 }
 
 
+/**
+ * Check if input is a valid url
+ * @param {Object} input
+ * @return {boolean} return true if url is found
+ */
+export function isValidModelURL(input) {
+  {/* eslint-disable-next-line */}
+  const urlRegex = '((http|https)://)(www.)?' + '[a-zA-Z0-9@:%._\\+~#?&//=]{2,256}\\.[a-z]' + '{2,6}\\b([-a-zA-Z0-9@:%._\\+~#?&//=]*)'
```

Comment:
Hehe, that's a crazy regex!  I would avoid using one of this complexity if there's a simpler way.  Either way please add tests for this method.

How about do the new URL parse and then access the .pathname field of it.. still a string and need a regex, but it will be much simpler.

Reply:
Added some tests. PTAL

Reply:
@pablo-mayrgundter 
Actually the new URL() give us this object:
<img width="518" alt="image" src="https://user-images.githubusercontent.com/3433606/163249137-9a5a16d5-a0dc-4192-8b67-08ac8918cbc6.png">

so i would need to check the hostname to be valid

Reply:
Agreed, but I think it's still preferable to have URL do the first parsing and then separate regexes for origin and pathname.  They'll be much simpler.  Regexes in general are hard so they should be short if possible.

Reply:
yes but this long regex is to validate the origin unless you can think of a shorter version, i goggled around and they all seemed long.
at the moment the pathname is not being validated, agree it should be.